packages/main/src/Menu.ts

```diff
@@ -676,9 +682,9 @@ class Menu extends UI5Element {
 		}
 	}
 
-	_findMainMenu(item: MenuItem) {
-		let parentMenu = item.parentElement as Menu;
-		while (parentMenu._parentMenuItem) {
+	_findMainMenu(element: MenuItem | Menu) {
+		let parentMenu = element instanceof Menu ? element : element.parentElement as Menu;
```

Comment:
same, don't use `instanceof`

Reply:
Done. Here we need to have the concern that the `ui5-navigation-menu` doesn't have `ui5-menu` attribute. So the same won't work there. Maybe the best approach would be to override the same method from inside the `ui5-navigation-menu`?

Reply:
Yes, in principle this is the expecation - that if you want `ui5-navigation-menu` to work with the same logic, you have to manually ensure it there - f.e. to override the function or to extract just this check in a separate small function and override only it for the navigation menu. But since we don't want this, and it doesn't break the navigation menu (Gery left a comment it's ok) then don't fix it there.